documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -30,67 +30,68 @@ We request OpenAPI(Swagger) spec authoring be assigned to engineers who have an
 | M3013	| DeleteMustNotHaveRequestBody | 'Delete' operation must not have a request body. | Error |
 | M2016	| PatchBodyParametersSchemaValidation | Properties of a PATCH request body must not be {0}. PATCH operation: '{1}' Model Definition: '{2}' Property: '{3}' | Error |
 | M2062	| PutResponseResourceValidation | The 200 response model for an ARM PUT operation must have x-ms-azure-resource extension set to true in its hierarchy. Operation: '{0}' Model: '{1}'. | Error |
-| [R3027](#R3027)	| TrackedResourceListByResourceGroup | The tracked resource, '{0}', must have a list by resource group operation. | Error |
-| [R3028](#R3028)	| TrackedResourceListBySubscription | The tracked resource, '{0}', must have a list by subscriptions operation. | Error |
-| [R3011](#R3011)	| DescriptionMustNotBeNodeName | The description provided for a given node (property, parameter, etc.) must not be the same as the name assigned to the node. | Error |
-| [R2020](#R2020) | RequiredPropertiesMissingInResourceModel | A `Resource` model must have `name`, `id` and `type` properties defined. | Error |
+| [R3027](#R3027)	| [TrackedResourceListByResourceGroup](#R3027) | The tracked resource, '{0}', must have a list by resource group operation. | Error |
+| [R3028](#R3028)	| [TrackedResourceListBySubscription](#R3028) | The tracked resource, '{0}', must have a list by subscriptions operation. | Error |
+| [R3011](#R3011)	| [DescriptionMustNotBeNodeName](#R3011) | The description provided for a given node (property, parameter, etc.) must not be the same as the name assigned to the node. | Error |
+| [R2020](#R2020) | [RequiredPropertiesMissingInResourceModel](#R2020) | A `Resource` model must have `name`, `id` and `type` properties defined. | Error |
 
 #### RPC Warnings
 
 | Id | Rule Name | Output Message |Severity | 
 | --- | --- | --- | --- |
 | M2061	| ProvidersPathValidation |	Type values \"{0}\" have default value(s), please consider parameterizing them | Warning |
-| [R3018](#R3018)	| BooleanPropertyNotRecommended	| Booleans are not descriptive and make them hard to use. Instead use string enums with allowed set of values defined. |	Warning |
-| [R3017](#R3017)	| GuidUsage | Guid used in model definition '{1}' for property '{0}'. Usage of Guid is not recommanded. If GUIDs are absolutely required in your service, please get sign off from the Azure API review board. | Warning |
-| M2057	| SkuModelValidation | Sku Model is not valid. A Sku model must have 'name' property. It can also have 'tier', 'size', 'family', 'capacity' as optional properties. | Warning |
-| [R3010](#R3010)	| TrackedResourceListByImmediateParent | The child tracked resource, '{0}' with immediate parent '{1}', must have a list by immediate parent operation. | Warning |
+| [R3018](#R3018)	| [EnumInsteadOfBoolean](#R3018)	| Booleans are not descriptive and make them hard to use. Consider using string enums with allowed set of values defined. Property: {0}. |	Warning |
```

Comment:
the rule is in production right now and the discussion hasn't closed on the ones to removed, therefore I included it. We can remove it, when removing the rule, if we do so.